opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -118,63 +118,118 @@ message MetricDescriptor {
-    GAUGE_INT64 = 1;
-
-    // Floating point gauge. The value can go both up and down over time.
-    // Corresponding values are stored in DoubleDataPoint.
-    GAUGE_DOUBLE = 2;
-
-    // Histogram gauge measurement.
-    // Used in scenarios like a snapshot of time that current items in a queue
-    // have spent there.
-    // Corresponding values are stored in HistogramDataPoint. The count and sum of the
-    // histogram can go both up and down over time. Recorded values are always >= 0.
-    GAUGE_HISTOGRAM = 3;
-
-    // Integer counter measurement. The value cannot decrease; if value is reset then
-    // start_time_unix_nano should also be reset.
-    // Corresponding values are stored in Int64DataPoint.
-    COUNTER_INT64 = 4;
-
-    // Floating point counter measurement. The value cannot decrease, if
-    // resets then the start_time_unix_nano should also be reset.
-    // Recorded values are always >= 0.
-    // Corresponding values are stored in DoubleDataPoint.
-    COUNTER_DOUBLE = 5;
-
-    // Histogram cumulative measurement.
-    // Corresponding values are stored in HistogramDataPoint. The count and sum of the
-    // histogram cannot decrease; if values are reset then start_time_unix_nano
-    // should also be reset to the new start timestamp.
-    CUMULATIVE_HISTOGRAM = 6;
+    // INVALID_TYPE is the default Type, it MUST not be used.
+    INVALID_TYPE = 0;
+
+    // INT64 values are represents as signed 64-bit integers.
+    //
+    // A Metric of this Type MUST store its values as Int64DataPoint.
+    INT64 = 1;
+
+    // DOUBLE values are represents as double-precision floating-point
+    // numbers.
+    //
+    // A Metric of this Type MUST store its values as DoubleDataPoint.
+    DOUBLE = 2;
+
+    // Histogram measurement.
+    // Corresponding values are stored in HistogramDataPoint.
+    HISTOGRAM = 3;
 
     // Summary value. Some frameworks implemented Histograms as a summary of observations
     // (usually things like request durations and response sizes). While it
     // also provides a total count of observations and a sum of all observed
     // values, it calculates configurable percentiles over a sliding time
     // window.
     // Corresponding values are stored in SummaryDataPoint.
-    SUMMARY = 7;
+    SUMMARY = 4;
   }
+
+  // type is the type of values this metric has.
   Type type = 4;
 
+  // Temporality is the temporal quality values of a metric have. It
+  // describes how those values relate to the time interval over which they
+  // are reported.
+  enum Temporality {
+    // INVALID_TEMPORALITY is the default Temporality, it MUST not be
+    // used.
+    INVALID_TEMPORALITY = 0;
+
+    // INSTANTANEOUS is a metric whose values are measured at a particular
+    // instant. The values are not aggregated over any time interval and are
+    // unique per timestamp. As such, these metrics are not expected to have
+    // an associated start time.
+    INSTANTANEOUS = 1;
```

Comment:
It'd be great if a clarification is added about `start_time_unix_nano`, `time_unix_nano` in `Int64DataPoint` etc. For INSTANTANEOUS, `start_time_unix_nano` can be null. But what should be the value for `time_unix_nano`?

Reply:
Same here with regards to this point is expected to be addressed in a future PR (https://github.com/MrAlias/opentelemetry-proto/commit/5cfae029570935d54ee1b1c903c31a5fb074c44e) that incrementally builds to #137.

Reply:
Thanks for clarifying! I will look forward to the upcoming PRs.